pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,557 @@
+									},
+								},
+								LivenessProbe: &corev1.Probe{
+									Handler: corev1.Handler{
+										HTTPGet: &corev1.HTTPGetAction{
+											Host: nodeLocal,
+											Path: "/health",
+											Port: intstr.FromInt(8080),
+										},
+									},
+									InitialDelaySeconds: int32(60),
+									TimeoutSeconds:      int32(5),
+								},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										MountPath: "/run/xtables.lock",
+										Name:      "xtables-lock",
+										ReadOnly:  false,
+									},
+									{
+										MountPath: "/etc/coredns",
+										Name:      "config-volume",
+									},
+									{
+										MountPath: "/etc/kube-dns",
+										Name:      "kube-dns-config",
+									},
+								},
+							},
+						},
+						Volumes: []corev1.Volume{
+							{
+								Name: "xtables-lock",
+								VolumeSource: corev1.VolumeSource{
+									HostPath: &corev1.HostPathVolumeSource{
+										Path: "/run/xtables.lock",
+										Type: &hostPathFileOrCreate,
+									},
+								},
+							},
+							{
+								Name: "kube-dns-config",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: "kube-dns",
+										},
+										Optional: pointer.Bool(true),
+									},
+								},
+							},
+							{
+								Name: "config-volume",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMap.Name,
+										},
+										Items: []corev1.KeyToPath{
+											{
+												Key:  configDataKey,
+												Path: "Corefile.base",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		vpa *autoscalingv1beta2.VerticalPodAutoscaler
+	)
+	utilruntime.Must(references.InjectAnnotations(daemonset))
+
+	if c.values.VPAEnabled {
+		vpaUpdateMode := autoscalingv1beta2.UpdateModeAuto
+		vpa = &autoscalingv1beta2.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Spec: autoscalingv1beta2.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "DaemonSet",
+					Name:       daemonset.Name,
+				},
+				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
+					UpdateMode: &vpaUpdateMode,
+				},
+				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
+					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{
+						{
+							ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
+							MinAllowed: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("50m"),
+								corev1.ResourceMemory: resource.MustParse("150Mi"),
+							},
+						},
+					},
+				},
```

Comment:
Nice, thanks for adding this!

Reply:
Should I change the values and keep CPU as 25m and memory and 25Mi, cause In daemonset  requested values are 

`Requests: corev1.ResourceList{
						corev1.ResourceCPU:    resource.MustParse("25m"),
						corev1.ResourceMemory: resource.MustParse("25Mi"),
						     },`

Reply:
Why `minAllowed` for the memory exceeds the memory limit configured in the DaemonSet? How you decided on these `minAllowed` values?

```suggestion
				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{
						{
							ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
							MinAllowed: corev1.ResourceList{
								corev1.ResourceCPU:    resource.MustParse("10m"),
								corev1.ResourceMemory: resource.MustParse("20Mi"),
							},
						},
					},
				},

```

Reply:
I was suggesting 25m cause `node-local-dns` do similar work as `coredns`.

Reply:
I guess `10m` should be fine as starting point.
On a relatively busy cluster the limit is set to `5m` and the actual usage is more or less within the limit (except a small throttle factor `1.15`).

![Screenshot 2021-12-20 at 12 59 31](https://user-images.githubusercontent.com/9372594/146757270-791796c2-b8c0-4fa8-a523-b09ae3072825.png)

Reply:
Ok, Thanks